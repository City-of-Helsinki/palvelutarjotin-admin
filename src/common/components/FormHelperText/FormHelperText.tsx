import classnames from 'classnames';
import React from 'react';

import styles from './formHelperText.module.scss';

const FormHelperText: React.FC<{
  text: string;
  className?: string;
  style?: React.CSSProperties;
}> = ({ text, className, style }) => {
  return (
    <div className={classnames(styles.helperText, className)} style={style}>
      {text}
    </div>
  );
};

export default FormHelperText;
