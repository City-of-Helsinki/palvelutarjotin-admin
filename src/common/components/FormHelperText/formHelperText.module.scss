.helperText {
  composes: hds-text-input from '~hds-core/lib/components/text-input/text-input.css';
  composes: hds-text-input__helper-text from '~hds-core/lib/components/text-input/text-input.css';
  margin: 0;
}
