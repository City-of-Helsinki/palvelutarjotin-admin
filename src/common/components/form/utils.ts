/* eslint-disable @typescript-eslint/no-explicit-any */
import { FormikErrors, FormikTouched } from 'formik';
import { TFunction } from 'i18next';
import get from 'lodash/get';

/** Get error text
 * @param {Object} errors
 * @param {Object} touched
 * @param {string} name
 * @param {Function} t
 * @return {string}
 */
export const getErrorText = (
  errors: FormikErrors<any>,
  touched: FormikTouched<any>,
  name: string,
  t: TFunction
) => {
  const error: any = get(errors, name);

  return !!error && get(touched, name)
    ? typeof error === 'string'
      ? (t(error) as string)
      : (t(error.key, error) as unknown as string)
    : '';
};
