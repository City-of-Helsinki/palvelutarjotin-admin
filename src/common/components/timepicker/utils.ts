export type TimeObject = { hours: number; minutes: number };

export function formatTime({ hours, minutes }: TimeObject): string {
  const hour = hours.toString().padStart(2, '0');
  const minute = minutes.toString().padStart(2, '0');
  return `${hour}:${minute}`;
}

export function getTimes(interval = 15): string[] {
  const times = [];

  for (let h = 0; h < 24; h += 1) {
    for (let m = 0; m < 60; m += interval) {
      times.push(formatTime({ hours: h, minutes: m }));
    }
  }

  return times;
}

export function getTimeObjects(interval = 15): TimeObject[] {
  const times: TimeObject[] = [];

  for (let h = 0; h < 24; h += 1) {
    for (let m = 0; m < 60; m += interval) {
      times.push({ hours: h, minutes: m });
    }
  }

  return times;
}
