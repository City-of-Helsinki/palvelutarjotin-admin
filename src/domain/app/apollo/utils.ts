import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
