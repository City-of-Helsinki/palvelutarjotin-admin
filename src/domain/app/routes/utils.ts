import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
