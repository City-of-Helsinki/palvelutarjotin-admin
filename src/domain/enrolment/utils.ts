import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
