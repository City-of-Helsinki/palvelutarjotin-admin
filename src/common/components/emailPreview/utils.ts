import { TEMPLATE_TYPE } from '../../../domain/enrolment/types';
import { EmailTemplatesQuery } from '../../../generated/graphql';
import { Language } from '../../../types';

export const getTranslatedTemplate = (
  templates: EmailTemplatesQuery | undefined,
  templateType: TEMPLATE_TYPE,
  locale: Language
) => {
  if (!templates) return null;

  const template = templates.notificationTemplates?.edges.find(
    (t) => t?.node?.type === templateType
  );
  const translatedTemplate = template?.node?.translations.find(
    (i) => i?.languageCode.toLowerCase() === locale
  );

  return translatedTemplate;
};
