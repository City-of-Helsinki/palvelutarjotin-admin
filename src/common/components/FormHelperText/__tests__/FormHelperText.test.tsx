import * as React from 'react';

import { customRender } from '../../../../utils/testUtils';
import FormHelperText from '../FormHelperText';

test('matches snapshot', () => {
  const { container } = customRender(
    <FormHelperText text="Test" className="className" />
  );

  expect(container).toMatchSnapshot();
});
