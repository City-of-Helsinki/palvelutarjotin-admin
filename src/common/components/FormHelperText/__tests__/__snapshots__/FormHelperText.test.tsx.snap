// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<div>
  <div
    class="helperText className"
  >
    Test
  </div>
</div>
`;
